lib/pharos/configuration.rb

```diff
@@ -0,0 +1,12 @@
+require 'dry-struct'
+
+module Pharos
+  module Configuration
+    class Struct < Dry::Struct
+      transform_types do |type|
+        # all attributes are optional and default to nil...
+        type.meta(omittable: true)
```

Comment:
I don't think we should be leaving all of these structs so weakly typed, but fixing all the optional attributes to be optional and/or fixing the specs to pass all required attributes would need work... plus duplicating more of the `ConfigSchema` `dry-validation` rules?